pkg/kubectl/sorting_printer.go

```diff
@@ -60,8 +61,28 @@ func (s *SortingPrinter) sortObj(obj runtime.Object) error {
 	if len(objs) == 0 {
 		return nil
 	}
+
+	if err := SortObjects(objs, s.SortField); err != nil {
+		return err
+	}
+
+	switch list := obj.(type) {
+	case *v1.List:
+		for ix := range objs {
+			data, err := v1.Codec.Encode(objs[ix])
```

Comment:
Yeah, that's not how the type is defined:

https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/types.go#L2224

Anyway, I worked around it a different way that doesn't require re-encoding the data.  It's hacky, but it works.